src/protocols/WebSocket.ts

```diff
@@ -148,8 +188,15 @@ export default class WebSocketProtocol extends BaseProtocolRealtime {
           const error = new KuzzleError(data.error);
           this.emit('queryError', error, data);
         }
+        /**
+         * In case you're running a Kuzzle version under 2.10.0
+         * The response from a browser custom ping will be an error
+         * since it does not parse the client message properly.
+         * We need to clear this timeout at each message to keep 
+         * the connection alive if it's the case
+         */
+        clearTimeout(this.pongTimeoutId);
```

Comment:
Shouldn't this be in the error handling `else` bloc according to the comment?

Also I don't understand the message, if we receive a message from Kuzzle we don't care if it's an error or not. Since we received a message then we should clean the pong timer and consider the connection as alive.

It seems that this is the actual behavior but then I think your comment is wrong.

Reply:
The comment was not accurate